File: LeetCode/Easy/The Two Sneaky Numbers of Digitville/3289.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var getSneakyNumbers = function(nums) {
    const result = [];
    const check = new Set();

    for (const num of nums) {
        if (check.has(num)) {
            result.push(num);
            if (result.length === 2) break;
        }
        check.add(num);
    }

    return result;
};

File: LeetCode/Easy/The Two Sneaky Numbers of Digitville/3289.java
class Solution {
    public int[] getSneakyNumbers(int[] nums) {
        int[] result = new int[2];
        boolean[] check = new boolean[nums.length];

        int i = 0;
        for (int num : nums) {
            if (check[num]) {
                result[i++] = num;
                if (i == 2) break;
            }
            check[num] = true;
        }

        return result;
    }
}
